src/sentry/tasks/integrations/link_all_repos.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import sentry_sdk
+
+from sentry.models.organization import Organization
+from sentry.plugins.base import bindings
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(name="sentry.integrations.github.link_all_repos", queue="integrations")
+def link_all_repos(
+    integration_key: str,
+    integration_id: int,
+    organization_id: int,
+):
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if not integration:
+        logger.error(
+            f"{integration_key}.link_all_repos.integration_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
+        return
+
+    try:
+        organization = Organization.objects.get(id=organization_id)
+    except Organization.DoesNotExist:
+        logger.error(
+            f"{integration_key}.link_all_repos.organization_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr(
+            f"{integration_key}.link_all_repos.error",
+            tags={"type": "missing_organization"},
+        )
+        return
+
+    installation = integration_service.get_installation(
+        integration=integration, organization_id=organization_id
+    )
+
+    client = installation.get_client()
+
+    try:
+        repositories = client.get_repositories(fetch_max_pages=True)
+    except ApiError as e:
+        if installation.is_rate_limited_error(e):
+            return
+
+        metrics.incr(f"{integration_key}.link_all_repos.api_error")
+        raise e
+
+    binding_key = "integration-repository.provider"
+    provider_key = (
+        integration.provider
+        if integration.provider.startswith("integrations:")
+        else "integrations:" + integration.provider
+    )
+    provider_cls = bindings.get(binding_key).get(provider_key)
+    integration_repo_provider = provider_cls(id=provider_key)
+
+    for repo in repositories:
+        try:
+            config = {
```

Comment:
similarly, how we build this config could probably be a helper function or dataclass. Maybe a TypeDict?

Reply:
i'll pull it out into a helper function